### PR TITLE
Refactoring block sync tests

### PR DESCRIPTION
**Pull request by danokoch**
Renaming blocksync test names and refactoring the tests to make them more readable (move the When's to harness)

### CODE REVIEW

File: services/blockstorage/sync/harness.go [hunk 1]
```diff
@@ -168,8 +131,7 @@ func (h *blockSyncHarness) withBatchSize(size uint32) *blockSyncHarness {
 }
 
 func (h *blockSyncHarness) expectingSyncOnStart() {
```
Comment (talkol) [inline services/blockstorage/sync/harness.go]: Let's use consistent naming between `expect` and `expecting`

File: services/blockstorage/sync/harness.go [hunk 2]
```diff
@@ -168,8 +131,7 @@ func (h *blockSyncHarness) withBatchSize(size uint32) *blockSyncHarness {
...
-	h.storage.When("UpdateConsensusAlgosAboutLatestCommittedBlock", mock.Any).Times(1)
-	h.expectLastCommittedBlockHeight(primitives.BlockHeight(10))
+	h.expectPreSynchronizationUpdateOfConsensusAlgos(10)
 	h.gossip.When("BroadcastBlockAvailabilityRequest", mock.Any, mock.Any).Return(nil, nil).Times(1)
```
Comment (talkol) [inline services/blockstorage/sync/harness.go]: Let's use the function we have to replace this direct When

File: services/blockstorage/sync/harness.go [hunk 3]
```diff
@@ -12,49 +12,12 @@ import (
 	"github.com/orbs-network/orbs-spec/types/go/protocol/gossipmessages"
 	"github.com/orbs-network/orbs-spec/types/go/services"
 	"github.com/orbs-network/orbs-spec/types/go/services/gossiptopics"
+	"github.com/pkg/errors"
 	"github.com/stretchr/testify/require"
 	"testing"
 	"time"
 )
 
-// the storage mock should be moved to its own file, but i have a weird goland bug where it will not identify it and its driving me mad, putting this here for now
```
Comment (talkol) [inline services/blockstorage/sync/harness.go]: I assume the reason it didn't work last time was that the new file was named `storage_mock_test.go` and not `storage_mock.go`

Unfortunately we can't name it `_test`

File: services/blockstorage/sync/storage_mock.go [hunk 4]
```diff
@@ -0,0 +1,43 @@
+package sync
+
+import (
+	"context"
+	"github.com/orbs-network/go-mock"
+	"github.com/orbs-network/orbs-spec/types/go/services"
+)
+
+// the storage mock should be moved to its own file, but i have a weird goland bug where it will not identify it and its driving me mad, putting this here for now
```
Comment (jlevison) [inline services/blockstorage/sync/storage_mock.go]: this comment can go to a better place now :) thanks for moving it.

**Pull request merged**